crates/biome_js_analyze/src/lint/nursery/no_flat_map_identity.rs

```diff
@@ -0,0 +1,141 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{ident, js_call_argument_list, js_call_arguments, js_name, token};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsFunctionBody, AnyJsMemberExpression, AnyJsName, AnyJsStatement,
+    JsCallExpression, JsSyntaxKind,
+};
+use biome_rowan::{AstNode, AstSeparatedList, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow to use unnecessary callback on `flatMap`,
+    ///
+    /// to achieve the same result (flattening an array) more concisely and efficiently, you should use `flat` instead
```

Comment:
```suggestion
    /// To achieve the same result (flattening an array) more concisely and efficiently, you should use `flat` instead
```

Reply:
@isnakode FYI: You can use `Commit suggestion` feature on GitHub, this feature prevents forgetting to address, but we need `just gen-***` after applying in most cases.

Reply:
oh thanks, does this feature auto commit to repo's fork? so i can pull it locally?

Reply:
It commits to this PR, yes you also need to pull this pr commits locally (and execute just command depends on the changed files)